packages/igx-templates/Update.ts

```diff
@@ -1,10 +1,36 @@
 import { App, FS_TOKEN, IFileSystem, PackageManager, ProjectConfig, Util } from "@igniteui/cli-core";
 import * as path from "path";
-import { FEED_PACKAGE, NPM_PACKAGE, resolvePackage } from "./package-resolve";
+import { FEED_DOCK_MANAGER, FEED_PACKAGE, NPM_DOCK_MANAGER, NPM_PACKAGE, resolvePackage } from "./package-resolve";
 
 const styleExtension = ["css", "scss", "sass"];
-const styleExpression = String.raw`(node_modules\/|~)${NPM_PACKAGE}(\/)`;
-const logicExpression = String.raw`(from ["'])${NPM_PACKAGE}(["'])`;
+enum RegularExpressionType {
+	STYLE,
+	LOGIC
+}
+
+// tslint:disable: object-literal-sort-keys
+const packages: { [key: string]: { trial: string, licensed: string, styleExp: string, logicExp: string } } = {
+	base: {
+		trial: NPM_PACKAGE,
+		licensed: FEED_PACKAGE,
+		styleExp: createExpression(RegularExpressionType.STYLE, NPM_PACKAGE),
+		logicExp: createExpression(RegularExpressionType.LOGIC, NPM_PACKAGE)
+	},
+	dockManager: {
+		trial: NPM_DOCK_MANAGER,
+		licensed: FEED_DOCK_MANAGER,
+		styleExp: createExpression(RegularExpressionType.STYLE, NPM_DOCK_MANAGER),
+		logicExp: createExpression(RegularExpressionType.LOGIC, NPM_DOCK_MANAGER)
+	}
+};
+
+function createExpression(expressionType: RegularExpressionType, packageName: string): string {
+	if (expressionType === RegularExpressionType.LOGIC) {
+		return String.raw`(from ["'])${packageName}(?<submodules>\/.*?)?(["'])`;
+	} else if (expressionType === RegularExpressionType.STYLE) {
+		return String.raw`(node_modules\/|~)${packageName}(\/)`;
+	}
+}
 
 export async function updateWorkspace(rootPath: string): Promise<boolean> {
 	if (resolvePackage() === FEED_PACKAGE) {
```

Comment:
This + the folder check for `@infragistics` are shaky at best, but will have to do for initial testing, especially now with the dock manager added. Will need a bit of re-work to handle both packages (separate resolves) and perhaps respect the root path.